File: 2006.Count_Number_of_Pairs_With_Absolute_Difference_K.java
class Solution {
    public int countKDifference(int[] nums, int k) {
        int count = 0, n = nums.length;
        for(int i=0;i<n;i++)
        {
            for(int j=i+1;j<n;j++)
            {
                if(Math.abs(nums[i]-nums[j]) == k)
                    count++;
            }
        }
        return count;
    }
    
}

File: 2006.Count_Number_of_Pairs_With_Absolute_Difference_K.cpp
class Solution {
public:
    int countKDifference(vector<int>& nums, int k) {
        unordered_map<int, int> map;
        int count = 0, n = nums.size();
        for(int i=0;i<n;i++)
        {
            if(map[nums[i]-k])
                count += map[nums[i]-k];
            if(map[nums[i]+k])
                count += map[nums[i]+k];
            
            map[nums[i]]++;
        }
        return count;
    }
};
